erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.js

```diff
@@ -68,52 +64,105 @@ erpnext.maintenance.MaintenanceSchedule = frappe.ui.form.Controller.extend({
 					});
 				}, __("Get Items From"));
 		} else if (this.frm.doc.docstatus === 1) {
-			this.frm.add_custom_button(__('Create Maintenance Visit'), function() {
-				frappe.model.open_mapped_doc({
-					method: "erpnext.maintenance.doctype.maintenance_schedule.maintenance_schedule.make_maintenance_visit",
-					source_name: me.frm.doc.name,
-					frm: me.frm
+			var s = me.frm.doc.schedules;
+			let flag = 0;
+			for (let i in s) {
+				if (s[i].completion_status == "Pending") {
+					flag = 1;
+				}
+			}
+			if (flag) {
+				this.frm.add_custom_button(__('Create Maintenance Visit'), function () {
+					let options = "";
+					
+					me.frm.call('get_pending_data',{data_type:"items"}).then(r =>{
+						options = r.message
+						
+					var schedule_id = "";
+					var d = new frappe.ui.Dialog({
+						title: __("Enter Visit Details"),
+						fields: [{
+							fieldtype: "Select",
+							fieldname: "item_name",
+							label: __("Item Name"),
+							options: options,
+							reqd: 1,
+							onchange: function () {
+								let field = d.get_field("scheduled_date");
+								me.frm.call('get_pending_data',{item_name:this.value,data_type:"date"}).then(r =>{
+									field.df.options = r.message;
+									field.refresh();
+								})
+							}
+						},
+						{
+							label: __('Scheduled Date'),
+							fieldname: 'scheduled_date',
+							fieldtype: 'Select',
+							options: "",
+							reqd: 1,
+							onchange: function () {
+								let field = d.get_field('item_name');
+								me.frm.call('get_pending_data',{item_name:field.value,s_date:this.value,data_type:"id"}).then(r =>{
+									schedule_id = r.message;
+								})
+							}
+						},
+						],
+						primary_action_label: 'Create Visit',
+						primary_action(values) {
+							frappe.call({
+								method: "erpnext.maintenance.doctype.maintenance_schedule.maintenance_schedule.make_maintenance_visit",
+								args: {
+									item_name: values.item_name,
+									s_id: schedule_id,
+									source_name: me.frm.doc.name,
+
+								},
+								callback: function (r) {
+									if (!r.exc) {
+										frappe.model.sync(r.message);
+										frappe.set_route("Form", r.message.doctype, r.message.name);
+									}
+								}
+
+
+							});
+							d.hide();
```

Comment:
dont leave extra spaces, the brackets in js are supposed to show where a function ends, not spaces